extensions/dicom-microscopy/src/DicomMicroscopyViewport.tsx

```diff
@@ -1,318 +1,217 @@
-        // TODO: refactor this after Bill's changes on ContextMenu feature get merged
-        // const roiAnnotationNearBy = this.getNearbyROI(event);
-      });
-    };
+        setViewer(viewer);
+        setManagedViewer(managedViewer);
+      };
 
-    this.microscopyService.clearAnnotations();
+      microscopyService.clearAnnotations();
 
-    let smDisplaySet = displaySet;
-    if (displaySet.Modality === 'SR') {
-      // for SR displaySet, let's load the actual image displaySet
-      smDisplaySet = displaySet.getSourceDisplaySet();
-    }
-    console.log('Loading viewer metadata', smDisplaySet);
+      let smDisplaySet = displaySet;
+      if (displaySet.Modality === 'SR') {
+        // for SR displaySet, let's load the actual image displaySet
+        smDisplaySet = displaySet.getSourceDisplaySet();
+      }
+      console.log('Loading viewer metadata', smDisplaySet);
 
-    await loadViewer(smDisplaySet.others);
+      await loadViewer(smDisplaySet.others);
 
-    if (displaySet.Modality === 'SR') {
-      displaySet.load(smDisplaySet);
-    }
-  }
+      if (displaySet.Modality === 'SR') {
+        displaySet.load(smDisplaySet);
+      }
+    },
+    [dataSource, extensionManager, microscopyService, servicesManager, viewportId]
+  );
 
-  componentDidMount() {
-    const { displaySets, viewportOptions } = this.props;
-    // Todo-rename: this is always getting the 0
+  useEffect(() => {
     const displaySet = displaySets[0];
-    this.installOpenLayersRenderer(this.container.current, displaySet).then(() => {
-      this.setState({ isLoaded: true });
+    installOpenLayersRenderer(container.current, displaySet).then(() => {
+      setIsLoaded(true);
     });
-  }
-
-  componentDidUpdate(prevProps: Readonly<{}>, prevState: Readonly<{}>, snapshot?: any): void {
-    if (this.managedViewer && prevProps.displaySets !== this.props.displaySets) {
-      const { displaySets } = this.props;
-      const displaySet = displaySets[0];
 
-      this.microscopyService.clearAnnotations();
-
-      // loading SR
-      if (displaySet.Modality === 'SR') {
-        const referencedDisplaySet = displaySet.getSourceDisplaySet();
-        displaySet.load(referencedDisplaySet);
+    return () => {
+      if (viewer) {
+        microscopyService.removeViewer(viewer);
       }
-    }
-  }
+    };
+  }, []);
 
-  componentWillUnmount() {
-    this.microscopyService.removeViewer(this.viewer);
-  }
+  useEffect(() => {
+    const displaySet = displaySets[0];
+
+    microscopyService.clearAnnotations();
 
-  setViewportActiveHandler = () => {
-    const { setViewportActive, viewportId, activeViewportId } = this.props;
+    // loading SR
+    if (displaySet.Modality === 'SR') {
+      const referencedDisplaySet = displaySet.getSourceDisplaySet();
+      displaySet.load(referencedDisplaySet);
+    }
+  }, [managedViewer, displaySets, microscopyService]);
 
+  function setViewportActiveHandler() {
```

Comment:
Could you inline the function logic below? I don't think we need to create a separate function for it here.